repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,254 @@
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        download_dir = directory_prefix
+    else:
+        download_dir = os.getcwd() + "/downloads"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+        )
+        updater.refresh()
+
+        info = updater.get_targetinfo(artifact_name)
+
+        if info is None:
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)
+        if path:
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)
+        console.print(f"Artifact downloaded and available in {path}")
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
+
+    return True
+
+
+@artifact.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.option(
+    "-P",
+    "--directory-prefix",
+    help="A prefix for the download dir.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.argument("artifact_name")
+@click.pass_context
+def download(
+    context: Context,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+    directory_prefix: str,
+    artifact_name,
+) -> None:
+    """
+    Downloads artifacts to the TUF metadata.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if context.obj.get("config") and os.path.isfile(context.obj.get("config")):
+        repository = rstuf_config.get("CURRENT_REPOSITORY")
+        # Overwriting the config if a flag is passed
+        # or using it as default instead
+        if not repository:
+            console.print("Please specify current repository")
+            sys.exit()
```

Comment:
Currently, when we have any error that causes a problem that we cannot resolve we throw an exception.
You can have a look at: https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/repository_service_tuf/cli/admin/ceremony.py, https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/repository_service_tuf/cli/admin/metadata.py, etc.

I think that will make for shorter code (as you don't need a `sys.exit()` after that.
Also, with this method you can easily forgot the `sys.exit()` call.

Finally, `sys.exit()` exits with a `0` code which is translated to a success.
That's not good especially if it's put in an automation somewhere.
I think you should change the uses of `sys.exit()` here in this code to exceptions to better show there is an actual problem that stops the execution.
Any thoughts @kairoaraujo?

Reply:
The initial idea of this was that as the CLI is not considered for automation purposes, but rather used interactively as an UX and when a user is missing a required parameter there is no actual error that happened during execution and a message without a failure is enough. I changed my mind though, as non-zero exit is the standard in such cases - coming from the idea that CLI tools exist for automation by default and, even though `rstuf` differs in that aspect, it's better to stick to the standard.

Reply:
That's something worth discussing with @kairoaraujo and @KAUTH as well.
How should we handle a situation where we don't have an argument we need and we cannot continue?
Do we raise an exception and return non-zero code or leave a warning message and exit?